pypesto/objective/petab_import.py

```diff
@@ -0,0 +1,363 @@
+        """
+        Create a pypesto.AmiciObjective.
+        """
+
+        condition_df = self.petab_problem.condition_df.reset_index()
+        measurement_df = self.petab_problem.measurement_df
+
+        # number of amici simulations will be number of unique
+        # (preequilibrationConditionId, simulationConditionId) pairs.
+        # Can be improved by checking for identical condition vectors.
+        grouping_cols, simulation_conditions = \
+            _get_simulation_conditions(condition_df, measurement_df)
+
+        observable_ids = self.model.getObservableIds()
+
+        fixed_parameter_ids = self.model.getFixedParameterIds()
+
+        edatas = []
+        for _, condition in simulation_conditions.iterrows():
+            # amici.ExpData for each simulation
+
+            # extract rows for condition
+            cur_measurement_df = _get_rows_for_condition(
+                measurement_df, grouping_cols, condition)
+
+            # make list of all timepoints for which measurements exist
+            timepoints = sorted(
+                cur_measurement_df.time.unique().astype(float))
+
+            # init edata object
+            edata = amici.ExpData(self.model.get())
+            edata.setTimepoints(timepoints)
+
+            # handle fixed parameters
+            if len(fixed_parameter_ids) > 0:
+                fixed_parameter_vals = condition_df.loc[
+                    condition_df.conditionId ==
+                    condition.simulationConditionId,
+                    fixed_parameter_ids].values
+                edata.fixedParameters = fixed_parameter_vals.astype(
+                    float).flatten()
+
+                if 'preequilibrationConditionId' in condition \
+                        and condition.preequilibrationConditionId:
+                    fixed_preequilibration_parameter_vals = condition_df.loc[
+                        # TODO: preequilibrationConditionId might not exist
+                        condition_df.conditionId == \
+                        condition.preequilibrationConditionId,
+                        fixed_parameter_ids].values
+                    edata.fixedParametersPreequilibration = \
+                        fixed_preequilibration_parameter_vals.astype(float) \
+                                                             .flatten()
+
+            # prepare measurement matrix
+            y = np.full(shape=(edata.nt(), edata.nytrue()), fill_value=np.nan)
+            # prepare sigma matrix
+            sigma_y = np.full(
+                shape=(edata.nt(), edata.nytrue()),
+                fill_value=np.nan)
+
+            # add measurements and sigmas
+            for _, measurement in cur_measurement_df.iterrows():
+                time_ix = timepoints.index(measurement.time)
+                observable_ix = observable_ids.index(
+                    f'observable_{measurement.observableId}')
+                # TODO: measurement file should contain prefix
```

Comment:
Should or shouldn't? I think we agreed to leave "observable_" out there.

Reply:
that's your comment :) . Yes, I also think that we agreed to leave it out now. I'll remove the comment.